Provide AppGroup directory as location for iOS

be-ndee opened the pull request:
This changes provide a new iOS Location called "Shared" which uses the AppGroup-ID to access a shared directory. If you set the `iosDatabaseLocation` to `Shared` and you set the preference `SqliteExtAppGroupId` in the `config.xml` to your AppGroup-ID, than the database can be used in the shared directory for your AppGroup.

I have to admit, that this is a very "fast" and simple way of adding this functionality, so I'm not sure if it's the right way. But it's working for me and I need it in a project.

be-ndee committed:
Provide shared-location for app groups

be-ndee committed:
change preference name

brody2consult commented:
Thanks @be-ndee for the nice contribution. In general I like the way this enhancement was coded.

That said, I would probably rework it in one of the following ways:
- shared group ID as an option in the openDatabase call, which would be given instead of location or iosDatabaseLocation
- iOS database directory URL in openDatabase call, directory URL would come from another plugin

For the second alternative there should probably be another plugin to lookup the directory path URL, based on the shared group ID.

brody2consult commented:
P.S. Some alternatives how to get Cordova app preferences from config.xml now tracked in <https://github.com/brodybits/cordova-notes/issues/2>
> Alternatives so far:
> - <https://github.com/apla/me.apla.cordova.app-preferences>
> - <https://taivo.github.io/guides/read-config-xml-in-cordova>
> - <https://github.com/rehy/cordova-plugin-shared-preferences> (for Android only, looks old)
> - <https://github.com/chrisekelley/AppPreferences> (second level fork, also looks old)

Commit:
use directory url instead of group id

be-ndee committed:
fix variable

be-ndee committed:
use correct variables and small fixes

be-ndee committed:
use correct function

be-ndee commented:
I've added some changes. I removed the "Shared" location and provided a new option called `iosDirectoryURL` where you can set the directory used in iOS. In the Objective-C code, the `iosDirectoryURL` and `dbName` are concatenated.

brody2consult commented:
Thanks @be-ndee for what looks like a wonderful contribution. I will probably need 1-2 weeks to test and integrate due to some other, pressing issues.

Commit:
set iosDirectoryLocation in deleteDatabase arguments

be-ndee committed:
set iosDirectoryLocation in deleteDatabase arguments

Commit:
Merge branch 'cordova-sqlite-ext-master' of github.com:be-ndee/cordova-sqlite-ext into cordova-sqlite-ext-master

be-ndee commented:
@brodybits what's the current progress of this pr?

brody2consult commented:
It may be another 1-2 months due to some client support needs and other pressing issues. My apologies.

brody2consult commented:
Please feel free to maintain and publish your own fork until I get a chance to deal with this one. I will likely support the arbitrary database location for all 3 mobile platforms Android/iOS/Windows at the same time, in a new major release.

be-ndee commented:
Thanks for the fast reply, you don't need to apologize. Just wanted to know, what's up at the moment ;)

Good idea to do it in a major release, which is more tested and not a fast-fix, like I did it ;)

Thank you anyway! :)

raphrmx commented:
Hello and congratulate for your work,

Would it be possible to get a functional version for ios shared app group?
I try to run this version but I can not :(

this is my code but my database is not in the shared app group

this.db = sqlitePlugin.openDatabase({name: 'tich.sqlite', iosDirectoryURL:'group.be.tich'})

With the master version, it works but the database is not in the group folder.

With the contrib-ios-shared1 version it gives me an error. I need to open the database before executing a transaction. Yet she is open.

Do you have an idea of the version I need to use and the settings I need to use?

Thank you so match in advance,
Raph :-)

be-ndee committed:
Remove not used location
